### PR TITLE
proposes alt+up and alt+down as a keybinding for moving between chunks

### PR DESCRIPTION
**Pull request by hamilton**
This PR implements the next / previous chunk keybinding suggested in #1421. For this PR, I implemented `alt+up` / `alt+down` since it doesn't seem to clash with any other keybindings to my knowledge, but I'm pretty indifferent to what we end up using, as long as it is easy on the hands.

### CODE REVIEW

File: src/editor/actions/actions.js [hunk 1]
```diff
@@ -204,6 +204,14 @@ export function updateEditorCursor(line, col, forceUpdate = false) {
   return { type: "UPDATE_CURSOR", line, col, forceUpdate };
 }
 
+export function moveToNextChunk() {
```
Comment (bcolloran) [inline src/editor/actions/actions.js]: there is already an action `moveCursorToNextChunk`

File: src/editor/reducers/notebook-reducer.js [hunk 2]
```diff
@@ -329,6 +329,38 @@ const notebookReducer = (state = newNotebook(), action) => {
       });
     }
 
+    case "MOVE_TO_NEXT_CHUNK": {
```
Comment (bcolloran) [inline src/editor/reducers/notebook-reducer.js]: you shouldn't need to touch the reducer to get this done -- take a look at the actions
`updateEditorCursor`and `moveCursorToNextChunk`

File: src/editor/actions/actions.js (hunk 1 above)
Comment (hamilton) [reply src/editor/actions/actions.js]: oh, thanks for the heads up. Of course there is! :|

File: src/editor/actions/__tests__/cursor-actions.test.js [hunk 3]
```diff
@@ -176,3 +180,151 @@ describe("moveCursorToNextChunk dispatches correct actions", () => {
     expect(store.getActions()).toEqual(expectedActions);
   });
 });
+
+describe("moveCursorToPreviousChunk dispatches correct actions", () => {
```
Comment (hamilton) [inline src/editor/actions/__tests__/cursor-actions.test.js]: this test mirrors the `moveCursorToNextChunk` tests, updated appropriately.

**Comment by hamilton** [comment]
@bcolloran thanks for the review. I've updated to use the action already there, and added another action - `moveCursorToPreviousChunk` + added tests. I've QA'd this locally against a handful of different notebooks too, just in case the tests aren't thorough enough.

If this all checks out, I'm happy to wait until we have more clarity on the right keybinding.

File: src/editor/actions/actions.js [hunk 4]
```diff
@@ -218,16 +218,45 @@ export function moveCursorToNextChunk() {
       jsmdChunks,
       editorCursor
     } = getState();
+    const lastChunk = jsmdChunks[jsmdChunks.length - 1];
+    if (editorCursor.line === lastChunk.endLine) {
+      dispatch(updateEditorCursor(lastChunk.endLine, Infinity, true));
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (hamilton) [reply src/editor/actions/actions.js]: sure, will add a test case. I think it depends on what should happen if you're already on the last chunk. Should the cursor just go to the end of the entire doc? If so, CodeMirror reads Infinity and places at the end.

[the remark this answers is not in the document]
Comment (hamilton) [reply src/editor/actions/actions.js]: I'll verify this bit though if this seems confusing. Before this PR the function was throwing an error at the edges of the document so the additional check was needed.

[the remark this answers is not in the document]
Comment (bcolloran) [reply src/editor/actions/actions.js]: i think it should just go down to the front of the last line rather than the end of the line

**Comment by bcolloran** [comment]
this is now getting kind of stale, and the tests will conflict with the monaco stuff (which uses 1-based indexing for buffer lines). i propose we close, since it will be quick to do again and it hasn't been touched in two months.

**Comment by hamilton** [comment]
Yep feel free to close. It was meant as a proof of concept, not as a
mergeable feature. The work isn't really lost if its closed.

**Comment by bcolloran** [comment]
(fwiw: i would have been happy to merge this! iirc, you consulted with a bunch of people and found this kbd shortcut to be the best option, right? we can pick it back up with monaco)

**Pull request closed without merging**